core/modules/widgets/navigator.js

```diff
@@ -557,10 +557,14 @@ NavigatorWidget.prototype.handleImportTiddlersEvent = function(event) {
 NavigatorWidget.prototype.handlePerformImportEvent = function(event) {
 	var self = this,
 		importTiddler = this.wiki.getTiddler(event.param),
-		importData = this.wiki.getTiddlerDataCached(event.param,{tiddlers: {}}),
+		importData,
 		importReport = [];
-	// Add the tiddlers to the store
+
 	importReport.push($tw.language.getString("Import/Imported/Hint") + "\n");
+	// If you need to modify the import tiddler payload, consider th-importing-tiddler instead!
+	importTiddler = $tw.hooks.invokeHook("th-before-importing",importTiddler);
+	// Update importData, if a hook did manipulate $:/Import 
```

Comment:
@pmario we are assigning importData and not updating it, so the comment is confusing. I think we can skip this comment entirely as the flow seems rather intuitive.